pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,34 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     $retryCount = 20,
+     $retryDelaySeconds = 10
+ 
+     $attempt = 1
+ 
+     while ($attempt -le $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $attempt returned null. Retrying in $retryDelaySeconds seconds..."
+             } else {
+                 Write-Host "Found HNS_NETWORK on attempt $($attempt + 1)"
```

Comment:
@wgahnagl   I think you can now remove the `$($attempt + 1)` , otherwise the attempt counter will start at `2` = `(1+1)`